apps/cowswap-frontend/src/legacy/state/price/updater.ts

```diff
@@ -79,25 +92,43 @@ function quoteUsingSameParameters(currentParams: FeeQuoteParams, quoteInfo: Quot
     hasSameReceiver
   // 2 checks: if there's a quoteInfo user address (meaning quote was already calculated once) and one without
   // in case user is not connected
-  return userAddress ? currentUserAddress === userAddress && paramsWithoutAddress : paramsWithoutAddress
+
+  const sameParams = userAddress ? currentUserAddress === userAddress && paramsWithoutAddress : paramsWithoutAddress
+
+  return {
+    fastQuoteSameParams: sameParams,
+    bestQuoteSameParams: sameParams && priceQuality === currentPriceQuality,
```

Comment:
Now the best quote uses the priceQuality, so I return the check as two independent flags